### PR TITLE
Add Dockerfile, refactor tests

### PR DESCRIPTION
**Pull request by mryab**

### CODE REVIEW

File: docs/user/benchmarks.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ hivemind.
 
 ### Server throughput
 
-You can use [this benchmark](https://github.com/learning-at-home/hivemind/blob/master/tests/benchmark_throughput.py) to
+You can use [this benchmark](https://github.com/learning-at-home/hivemind/blob/master/benchmarks/benchmark_throughput.py) to
```
Comment (justheuristic) [inline docs/user/benchmarks.md:8]: Thanks, this move was long overdue :)

p.s. I've added a task to re-run the benchmark numbers to https://github.com/learning-at-home/hivemind/issues/232

**Review by justheuristic — APPROVED** [review]
request: merge this after stability fix (which is due today), also bump version

File: .github/workflows/push-docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+name: Push to Docker Hub
+
+on:
+  push:
+    branches: [ master ]
+    tags:
+      - "*.*.*"
+  pull_request:
```
Comment (borzunov) [inline .github/workflows/push-docker-image.yml:8]: AFAIU, the [pull_request](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request) event is triggered when the PR metainfo (comments/labels/assigned people) changes but is not triggered when new commits are pushed.

Is this the intended behavior?

**Review by borzunov — CHANGES_REQUESTED** [review]
See the comment above.

File: .github/workflows/push-docker-image.yml (hunk 2 above)
Comment (mryab) [reply .github/workflows/push-docker-image.yml:8]: As you can see from the history of this PR, the event is indeed triggered on every push :) The [synchronize](https://github.community/t/what-is-a-pull-request-synchronize-event/14784/4) event is triggered on each push to the branch of PR

Comment (borzunov) [reply .github/workflows/push-docker-image.yml:8]: Got it, thanks for the explanation!

**Review by borzunov — APPROVED** [review]

**Pull request merged**